digits/config/caffe_option.py

```diff
@@ -141,78 +145,143 @@ def validate_version(cls, executable):
+            if NVIDIA_SUFFIX in filename:
+                CaffeOption.flavor = 'NVIDIA'
             else:
-                libname = 'libcaffe'
-                caffe_line = None
-
-                # Search output for caffe library
-                for line in p.stdout:
-                    if libname in line:
-                        caffe_line = line
-                        break
-                if caffe_line is None:
-                    raise config_option.BadValue('%s not found in ldd output' % libname)
-
-                # Read the symlink for libcaffe from ldd output
-                symlink = caffe_line.split()[2]
-                filename = os.path.basename(os.path.realpath(symlink))
-
-                # Check for the nvidia suffix
-                if NVIDIA_SUFFIX not in filename:
-                    raise config_option.BadValue('Library at "%s" does not have expected suffix "%s". Are you using the NVIDIA/caffe fork?'
-                            % (filename, NVIDIA_SUFFIX))
+                CaffeOption.flavor = 'BVLC'
 
+            if CaffeOption.version_string:
+                version_string = CaffeOption.version_string
+            else:
                 # parse the version string
-                match = re.match(r'%s%s\.so\.(\S+)$'
-                        % (libname, NVIDIA_SUFFIX), filename)
+                match = re.match(r'%s(.*)\.so\.(\S+)$'
+                                 % (libname), filename)
                 if match:
-                    version_str = match.group(1)
-                    return parse_version(version_str)
-                else:
-                    return None
-
-        elif platform.system() == 'Darwin':
-            # XXX: guess and let the user figure out errors later
-            return parse_version(0,11,0)
-        elif platform.system() == 'Windows':
-            p = subprocess.Popen([executable,'-version'],
-                    stdout = subprocess.PIPE,
-                    stderr = subprocess.PIPE)
-            if p.wait():
-                raise config_option.BadValue(p.stderr.read().strip())
-            else:
-                pattern = 'version'
-                parsed_version = parse_version(0,11,0)
-                for line in p.stdout:
-                    if pattern in line:
-                        try:
-                            # if CAFFE for Windows is built with properly formatted version string
-                            parsed_version = parse_version(line[line.find(pattern)+len(pattern):].rstrip())
-                        except ValueError:
-                            # what follows "version" is "CAFFE_VERSION" or some unrecognized string
-                            pass
-                        finally:
-                            break
-                return parsed_version
+                    version_string = match.group(2)
+            return version_string
+        else:
+            raise UnsupportedPlatformError('Linked library in Linux only')
```

Comment:
That's not exactly true, is it? We could use `otool` on OSX, at least. How about: "Parsing of shared library version not supported on this platform"?